nntrainer/layers/cl_layers/rmsnorm_layer_cl.cpp

```diff
@@ -0,0 +1,391 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Thummala Pallavi <t.pallavi@samsung.com>
+ *
+ * @file        rmsnorm_layer_cl.cpp
+ * @date        11 June 2024
+ * @brief       This is RMSNorm layer for Neural Network with OpenCl
+ * implementation
+ * @see         https://github.com/nnstreamer/nntrainer
+ * @author      Thummala Pallavi <t.pallavi@samsung.com>
+ * @bug         No known bugs except for NYI items
+ *
+ */
+
+#include <common_properties.h>
+#include <layer_context.h>
+#include <lazy_tensor.h>
+#include <nntrainer_error.h>
+#include <node_exporter.h>
+#include <rmsnorm_layer_cl.h>
+#include <stdio.h>
+#include <util_func.h>
+
+std::string rmsnorm_cl_kernel_fp16_ =
+  R"(
+    #pragma OPENCL EXTENSION cl_khr_fp16 : enable
+    __kernel void rmsnorm_cl_fp16(__global const half* input,
+                       __global half* output,
+                       __global const half* gamma,
+                       half epsilon,
+                       int N, int C, int H, int W){
+    int n = get_global_id(0);
+    int c = get_global_id(1);
+    int h = get_global_id(2);
+
+    // Calculate base index for the current (n, c, h) slice
+    int index_base = ((n * C + c) * H + h) * W;
+
+    // Compute the mean square along the width dimension
+    while(index_base < N*C*H*W){
```

Comment:
Is it correct to compute the for loop until `N*C*H*W` wherever it starts from? (n, c, h according to global id)
I couldn't understand how this code parallelize the computation.
Are N,C,H,W set different according to workers?

Since the work_group_count is set to {1, 1, 1}, it seems there wasn't redundant computation for now.
(Please refer to @s-debadri 's comment)
> {1, 1, 1} will have only one work item on the GPU.
> 
> ```c++
> int n = get_global_id(0);
> int c = get_global_id(1);
> int h = get_global_id(2);
> ```
> 
> Above lines in the kernels will have `n = 0, c = 0, h = 0` always. Please check this once using `print` statements or different input sizes.

However, when we make more work_groups... does this code work as you intended?
Could you explain about this?

Reply:
Yes, have made the respective changes will update the PR by today

Reply:
Updated the Code please review @s-debadri @EunjuYang